### PR TITLE
Add time dependence to ExportCoordinates

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

Extend ExportCoordinates executable to support moving meshes

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [x] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [x] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [x] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by geoffrey4444** [comment]
Test failure is #2078 (unrelatedly to changes in this PR)

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (kidder) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp]: you don't need to specify false as it is the default.

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsvu** [comment]
Is it possible to keep a version of this executable that does not require an `Evolution` section in the input file? The executable was intended as a simple way of visualizing a domain creator, and I think we should keep that functionality. How about we make the executable with an evolution section in the input file `ExportMovingMeshCoordinates`?

**Comment by nilsleiffischer** [comment]
Is it possible to keep a version of this executable that does not require an `Evolution` section in the input file? The executable was intended as a simple way of visualizing a domain creator, and I think we should keep that functionality. How about we make the executable with an evolution section in the input file `ExportingMovingMeshCoordinates`?

**Comment by geoffrey4444** [comment]
> Is it possible to keep a version of this executable that does not require an `Evolution` section in the input file? The executable was intended as a simple way of visualizing a domain creator, and I think we should keep that functionality. How about we make the executable with an evolution section in the input file `ExportMovingMeshCoordinates`?

I’m fine doing this if that’s the consensus, but wouldn’t this mean some code duplication?

**Comment by nilsvu** [comment]
I was thinking a few compile-time switches in the Metavariables to turn the evolution part on or off

**Comment by geoffrey4444** [comment]
@nilsleiffischer we discussed on the Caltech call today, where I did my best to represent your point of view. The consensus emerged that we don't want to implement your compiler-switch suggestion, because we want to avoid adding more executables. 

Note that this executable still works on time-independent meshes! Just set the final time and initial time to be zero in the input file, so you don't get the same time-independent coordinates output repeatedly.

**Comment by nilsvu** [comment]
tbh if we want fewer executables we should do https://github.com/sxs-collaboration/spectre/issues/1815

If `ExportCoordinates` still works for non-moving meshes in a way that also a new student would be able to run it without confusion then I‘m ok with that.

**Review by kidder — DISMISSED** [review]

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp]: Please don't make this change from the explicit tuple to `auto`. Just remove the `bool` from the tuple.

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp:148]: Should `Actions::RunEventsAndTriggers` be the first action in this list? It is for our evolution execs

File: tests/InputFiles/ExportCoordinates/Input1D.yaml
Comment (nilsvu) [inline tests/InputFiles/ExportCoordinates/Input1D.yaml]: Do you change the default domain for a particular reason? If no, just keep it as-is except for the `TimeDependence`. Also please don't change the indentation. (same for the other input files)

File: tests/InputFiles/ExportCoordinates/Input1D.yaml
Comment (nilsvu) [inline tests/InputFiles/ExportCoordinates/Input1D.yaml]: What do I need to change in this input file to export a static domain (once)? Perhaps make one of the input files an example for that and/or add comments to the files. I think a person looking at these files shouldn't need to figure that out by themselves. If I set `PastTime: 0.` I currently get two observations. If I set it to -1 I get only one, but that's a non-obvious thing to do.

I suggest this:
- Move the `Evolution` section to the very bottom, after `EventsAndTriggers`.
- Add a comment here like "Set `PastTime: 0.` (or whatever it should be) to export a domain with no time dependence"

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (wthrowe) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp:148]: In the evolution executables the first AdvanceTime action is done as part of the InitializeTimeStepperHistory phase, which isn't being used here.  That should be fine as long as only Euler's method (AB1) is used.

RunEventsAndTriggers is only being used to terminate the executable here, so it makes sense to put it just after the observation event.  (ExportCoordinates might be better suited as an event itself.)

File: tests/InputFiles/ExportCoordinates/Input1D.yaml
Comment (wthrowe) [reply tests/InputFiles/ExportCoordinates/Input1D.yaml]: This is caused by strict vs. non-strict inequalities.  (`PastTime: 0` checks `t > 0`, not `t >= 0`.)  I suggest `Always`.

**Review by wthrowe — COMMENTED** [review]
Not a review, just explaining some weird time-related stuff.

**Comment by geoffrey4444** [comment]
@nilsleiffischer I missed your suggested changes until now...when you have a chance, it's ready for you to take a look at my fixup commit addressing them

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp]: Delete this line

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp]: I don't think this alias is used. Please remove unless I'm mistaken

File: tests/InputFiles/ExportCoordinates/Input2D.yaml
Comment (nilsvu) [inline tests/InputFiles/ExportCoordinates/Input2D.yaml]: In the 2d and 3d input sfile you're still changing the default domain and the indentation, please revert (at least make sure the indentation is correct, but if you have no reason to change the default domain then just revert the diff)

**Review by nilsvu — CHANGES_REQUESTED** [review]
Please squash, including the few comments below

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (geoffrey4444) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp]: When I remove line 34, I get this compiler error: 
~~~
/home/geoffrey/Codes/spectre/spectre/src/Executables/ExportCoordinates/ExportCoordinates.hpp:183:16: error: no member named 'register_derived_classes_with_charm' in namespace 'Parallel'
    &Parallel::register_derived_classes_with_charm<TimeStepper>,
~~~

File: tests/InputFiles/ExportCoordinates/Input2D.yaml
Comment (geoffrey4444) [reply tests/InputFiles/ExportCoordinates/Input2D.yaml]: Sorry I missed the indent fix...I've got that in the squash I'm about to push. There is a reason to change the domain: Disk and Shell don't support TimeDependence yet, and I wanted the input files to include time dependence for the test.

Comment (geoffrey4444) [reply tests/InputFiles/ExportCoordinates/Input2D.yaml]: I changed the domain to one of the appropriate dimension that supports TimeDependence (Disk and Shell don't yet). However, domains like Disk and Shell still work fine as long as you don't try to specify TimeDependence: in the input file.

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp]: I meant line 35, since you replaced the action with the one from `Evolution`

Comment (geoffrey4444) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp]: O, I misunderstood! My eye went to the line I changed. Fixed and squashed @nilsleiffischer

**Review by nilsvu — APPROVED** [review]

**Comment by geoffrey4444** [comment]
@kidder this is just waiting for a second approval..it would be great if you could take a look when you get a chance...but I realize you're busy, so I wonder if you'd prefer if we find a different second reviewer?

**Review by kidder — APPROVED** [review]

**Pull request merged**
